Improve daterange picker

stephenwan opened the pull request:
Sync with the latest date-range-picker in wfm project.

Features:
- inline mode with input fields
- inherit from ng-model to be more compatible with standard angular form validation
- use uib-datepicker instead of deprecated datepicker from the latest angular-ui-bootstrap (used in wfm)

blandine commented on kss-template/public/templates.js:
```diff
@@ -0,0 +1,147 @@
+angular.module('styleguide.templates', []).run(['$templateCache', function($templateCache) {
```
Maybe better to add this file in .ignore since it's generated ?

blandine commented:
Thanks @stephenwan great job, especially since the directives in wfm will be removed to use only the ones from the styleguide :)
I've tested the directive a bit and found a bug when it's used in pop up mode though. At the end of the gif i can't do anything and after a while chrome crashes.
![daterangepickerbug](https://cloud.githubusercontent.com/assets/1214949/13219566/17db6aa2-d971-11e5-8502-a6f5216fef1d.gif)

blandine commented on Gruntfile.js:
```diff
@@ -73,11 +82,16 @@
+
     grunt.loadNpmTasks('grunt-contrib-uglify');
+    grunt.loadNpmTasks('grunt-angular-templates');
+    grunt.loadNpmTasks('grunt-processhtml');
```
is grunt-processhtml needed ? It seems not used.

stephenwan commented:
Thank you @blandine, I will review the problems you've mentioned. By the way, the video is awesome! How did you make it?

stephenwan commented:
Hi @blandine , I have moved the compiled templates into the ```dist/``` folder so it will be ignored together with wfmdirective.min.js.

One thing that I would like to add is that in this implementation I have changed how we sync the colored date-range in two datepickers. Because of lack of support from UI-Bootstrap, we hacked to our desired behavior by assigning two new Date objects to `startDate` and `endDate` to force refreshing the datepickers. But this method doesn't work at all with the popup mode of the new `uib-datepicker`. So instead I simulate clicking the active date cells from UI to force refreshing the datepickers with much smaller impact. FYI.

blandine commented:
Thanks @stephenwan ! For making gif, @TheRockman found [this tool called screentogif](https://screentogif.codeplex.com), it's indeed good and useful :)

I'll review your changes as well as the implementation you mentionned today.
